File: JAVA/590. N-ary Tree Postorder Traversal.java
class Solution {
    public List<Integer> postorder(Node root) {
        ArrayList<Integer> list = new ArrayList<>();
        post(root,list);
        return list;
    }
    public void post(Node root,ArrayList<Integer> list )
    {
        if(root==null) return;
        for (Node child : root.children) post(child, list);
        list.add(root.val);
    }
}

File: C++/590. N-ary Tree Postorder Traversal.cpp
class Solution {
public:
vector<int> ans;
vector<int> postorder(Node* root) {
    nary(root);
    return ans;
}
void nary(Node* root){
    if(root==NULL)
        return;
    for(auto a:root->children)
        nary(a);
    ans.push_back(root->val);
 }
};
